libbeat/cmd/instance/beat.go

```diff
@@ -285,6 +286,226 @@ func NewBeat(name, indexPrefix, v string, elasticLicensed bool, initFuncs []func
 	return &Beat{Beat: b}, nil
 }
 
+// NewBeatReceiver creates a Beat that will be used in the context of an otel receiver
+func NewBeatReceiver(settings Settings, receiverConfig map[string]interface{}, consumer consumer.Logs) (*Beat, error) {
+	b, err := NewBeat(settings.Name,
+		settings.IndexPrefix,
+		settings.Version,
+		settings.ElasticLicensed,
+		settings.Initialize)
+	if err != nil {
+		return nil, err
+	}
+
+	b.Info.LogConsumer = consumer
+
+	// begin code similar to configure
+	if err = plugin.Initialize(); err != nil {
+		return nil, fmt.Errorf("error initializing plugins: %w", err)
+	}
+
+	b.InputQueueSize = settings.InputQueueSize
+
+	cfOpts := []ucfg.Option{
+		ucfg.PathSep("."),
+		ucfg.ResolveEnv,
+		ucfg.VarExp,
+	}
+
+	tmp, err := ucfg.NewFrom(receiverConfig, cfOpts...)
+	if err != nil {
+		return nil, fmt.Errorf("error converting to ucfg: %w", err)
+	}
+
+	cfg := (*config.C)(tmp)
+	if err := initPaths(cfg); err != nil {
+		return nil, fmt.Errorf("error initializing paths: %w", err)
+	}
+
+	// We have to initialize the keystore before any unpack or merging the cloud
+	// options.
+	store, err := LoadKeystore(cfg, b.Info.Beat)
+	if err != nil {
+		return nil, fmt.Errorf("could not initialize the keystore: %w", err)
+	}
+
+	if settings.DisableConfigResolver {
+		config.OverwriteConfigOpts(obfuscateConfigOpts())
+	} else {
+		// TODO: Allow the options to be more flexible for dynamic changes
+		config.OverwriteConfigOpts(configOpts(store))
+	}
+
+	instrumentation, err := instrumentation.New(cfg, b.Info.Beat, b.Info.Version)
+	if err != nil {
+		return nil, fmt.Errorf("error setting up instrumentation: %w", err)
+	}
+	b.Beat.Instrumentation = instrumentation
+
+	b.keystore = store
+	b.Beat.Keystore = store
+	err = cloudid.OverwriteSettings(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("error overwriting cloudid settings: %w", err)
+	}
+
+	b.RawConfig = cfg
+	err = cfg.Unpack(&b.Config)
+	if err != nil {
+		return nil, fmt.Errorf("error unpacking config data: %w", err)
+	}
+
+	if err := promoteOutputQueueSettings(&b.Config); err != nil {
+		return nil, fmt.Errorf("could not promote output queue settings: %w", err)
+	}
+
+	if err := features.UpdateFromConfig(b.RawConfig); err != nil {
+		return nil, fmt.Errorf("could not parse features: %w", err)
+	}
+	b.RegisterHostname(features.FQDN())
+
+	b.Beat.Config = &b.Config.BeatConfig
+
+	if name := b.Config.Name; name != "" {
+		b.Info.Name = name
+	}
+
+	if err := common.SetTimestampPrecision(b.Config.TimestampPrecision); err != nil {
+		return nil, fmt.Errorf("error setting timestamp precision: %w", err)
+	}
+
+	if err := configure.LoggingWithTypedOutputs(b.Info.Beat, b.Config.Logging, b.Config.EventLogging, logp.TypeKey, logp.EventType); err != nil {
+		return nil, fmt.Errorf("error initializing logging: %w", err)
+	}
+
+	// log paths values to help with troubleshooting
+	logp.Info(paths.Paths.String())
```

Comment:
Do we need a prefix here, for example "Path values: %s" to add a bit of context otherwise this will be a naked string, no?

Reply:
paths already adds some info:

https://github.com/elastic/elastic-agent-libs/blob/5104a902304586d07906b91e169983699fa46d82/paths/paths.go#L158

You end up with logs that look like:

```
{"log.level":"info","@timestamp":"2024-08-02T14:29:23.209Z","message":"Home path: [C:\\Program Files\\Elastic\\Agent\\data\\elastic-agent-b0c688\\components] Config path: [C:\\Program Files\\Elastic\\Agent\\data\\elastic-agent-b0c688\\components] Data path: [C:\\Program Files\\Elastic\\Agent\\data\\elastic-agent-b0c688\\run\\http\\metrics-monitoring] Logs path: [C:\\Program Files\\Elastic\\Agent\\data\\elastic-agent-b0c688\\components\\logs]","component":{"binary":"metricbeat","dataset":"elastic_agent.metricbeat","id":"http/metrics-monitoring","type":"http/metrics"},"log":{"source":"http/metrics-monitoring"},"service.name":"metricbeat","ecs.version":"1.6.0","log.origin":{"file.line":783,"file.name":"instance/beat.go"},"ecs.version":"1.6.0"}
```

I'd suggest leaving it as is.  This is what is already happening for "normal" beat startup.  That way the logs are consistent, which hopefully makes it easier on the humans when debugging.  If this was "new" I agree that more context would be better.